test/utils.cpp

```diff
@@ -348,6 +348,18 @@ TEST_CASE("strwidth()", "[utils]")
 	REQUIRE(utils::strwidth(utils::wstr2str(L"\uF91F")) == 2);
 }
 
+TEST_CASE("is_http_url()", "[utils]")
+{
+	REQUIRE(utils::is_http_url("http://foo.bar"));
+	REQUIRE(utils::is_http_url("https://foo.bar"));
+	REQUIRE(utils::is_http_url("http://"));
+	REQUIRE(utils::is_http_url("https://"));
+
+	REQUIRE_FALSE(utils::is_http_url("htt://foo.bar"));
+	REQUIRE_FALSE(utils::is_http_url("http:/"));
+	REQUIRE_FALSE(utils::is_http_url("foo://bar"));
```

Comment:
I'm not sure if it is really worth to have that many tests for such a simple function or if there would be better test cases, but here they are :wink:

Reply:
Yeah, 6 tests are plenty. Thanks! Please port them to Rust, too. This way, when we remove the C++ "shell" of the function, the actual code is still covered by tests.

Reply:
Ah that makes sense. Done.